src/lib.rs

```diff
@@ -389,44 +364,39 @@ mod tls {
                 }
             });
 
-            #[cfg(feature = "tracing")]
-            {
-                tracing::debug!("Certificates loaded (Count: {})", certs.len());
-            }
-
             Ok(CertVerifier { certs: certs })
         }
     }
 
-    impl rustls::ServerCertVerifier for CertVerifier {
+    impl ServerCertVerifier for CertVerifier {
         fn verify_server_cert(
             &self,
-            _roots: &RootCertStore,
-            presented_certs: &[Certificate],
-            _dns_name: DNSNameRef<'_>,
+            end_entity: &Certificate,
+            intermediates: &[Certificate],
+            _server_name: &ServerName,
+            _scts: &mut dyn Iterator<Item = &[u8]>,
             _ocsp_response: &[u8],
+            _now: SystemTime,
         ) -> Result<ServerCertVerified, TLSError> {
-            if self.certs.len() != presented_certs.len() {
+            if self.certs.len() != intermediates.len() + 1 {
                 return Err(TLSError::General(format!(
                     "Mismatched number of certificates (Expected: {}, Presented: {})",
                     self.certs.len(),
-                    presented_certs.len()
+                    intermediates.len() + 1
                 )));
             }
-
-            for (c, p) in self.certs.iter().zip(presented_certs.iter()) {
+            for (c, p) in self.certs.iter().zip(intermediates.iter()) {
                 if *p.0 != **c {
                     return Err(TLSError::General(format!(
                         "Server certificates do not match ours"
                     )));
-                } else {
-                    #[cfg(feature = "tracing")]
-                    {
-                        tracing::trace!("Confirmed certificate match");
-                    }
                 }
             }
-
+            if let Some(last) = self.certs.last() {
+                if *last != end_entity.0 {
```

Comment:
This is the only part I'm not 100% sure about, it would be good to double-check for a cert that is signed by letsencrypt

Reply:
I've dropped this custom validation after checking against letsencrypt certs and seeing failures. My tests were inconclusive though. Some of the failures seemed expected, i.e when I passed a certificate that was not provided by server to client

Maybe it's best to consider the whole custom validation changes as follow up improvement

Reply:
How did you manage to create the letsencytp cert for LND? I'll give it a go tomorrow and if we cannot get far we can just leave it as it was and tackle it down the line

Reply:
I have a signet lnd with letsencrypt certs already set up. Let's test it together tomorrow